### ISSUE TITLE
L-BFGS Optimizer & TPU support?

### ISSUE THREAD

**Issue by JEF1056** [lengstrom/fast-style-transfer#167]
I really like the project, but I really would like to see optimizations made for faster style with the L-BFGS optimizer; I know that it's been done on torch, but a tensorflow example would be extremely helpful ^-^

Also any idea where the tf.session() is? I really want to try training this on TPU

**Comment by JEF1056** [comment]
I'm working on adding the L-BFGS optimizer in my own fork of this repo, if I can get it working and lengstrom allows me, I'll transfer the code to this repo
https://github.com/JEF1056/fast-style-transfer

**Comment by JEF1056** [comment]
@lengstrom It would be great if you could review the L-BFGS optimizer changes I tried to merge into this repo

**Comment by NotRobert** [comment]
@JEF1056 : 
What is the status if your fork? Is your L-BFGS implementation working? If so, what enhancements do you have?

**Comment by JEF1056** [comment]
@NotRobert 
Currently, it works on GPU only, and benefits are very small; in testing, I've observed a complete lack of grey spot areas, and less grain in the resulting image. However, more testing needs to be done before I can say that it's really usable or better than the ADAM implementation.
